### DISCUSSION TITLE
Add operation name to subscribe message

### DISCUSSION THREAD

**Discussion by tomas-light** [enisdenjo/graphql-ws#440] [Q&A] [answered]
### Story

As a client

I want to fast navigate in many messages in browser Network tab for web socket like:

![image](https://user-images.githubusercontent.com/20392714/214828354-bc19646d-1ebb-48e5-a22a-28830d395812.png)

You can see, there are a lot of messages sent via web socket, and without operation name it is too hard to observe all of them and find the needed one for debug

So I prepare code adjustments for that purpose:
https://github.com/enisdenjo/graphql-ws/compare/master...TomasLight:graphql-ws:master

**Comment by enisdenjo** [comment]
~~Similar to #398.~~

~~I wouldn't add a new field to the subscribe message, I'd recommend instead using the [`generateID` client option](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/client.ClientOptions.md#generateid) to ease up lookups during development. The argument receives the operation payload so you can construct whatever you want.~~

Moved to https://github.com/enisdenjo/graphql-ws/discussions/440#discussioncomment-4816827.

**Comment by tomas-light** [comment]
@enisdenjo, what if I will have multiple messages with same ID (operation name)? Will it affect any internal implementations (maybe sync requests and responses or similar)?

**Comment by enisdenjo** [comment, marked as answer]
Similar to #398.

I wouldn't add a new field to the subscribe message, I'd recommend instead using the [`generateID` client option](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/client.ClientOptions.md#generateid) to ease up lookups during development. The argument receives the operation payload so you can construct whatever you want.

You can add something random to disambiguate the IDs during development. Roughly:

```js
import { createClient } from 'graphql-ws';

const client = createClient({
  url: 'ws://dev-gen-id:4000/graphql',
  generateID: ({ operationName }) =>
    // operationName plus a random 4 character hash
    `${operationName}_${Math.random().toString(36).slice(-4)}`,
});
```

Do note that the [GraphQL over WebSocket Protocol disallows multiple active operations with the same ID](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md#subscribe).
